FWCore/Framework/interface/ESProducer.h

```diff
@@ -242,6 +241,105 @@ namespace edm {
       return ESConsumesCollectorT<TRecord>(consumesInfos_.back().get(), id);
     }
 
+    // These replicate the setWhatProduced functions but add a second functor for
+    // the acquire step. The acquire step is analogous to the acquire step for
+    // EDProducers with the ExternalWork ability.
+
+    template <typename T>
+    auto setWhatProducedExternalWork(T* iThis, const es::Label& iLabel = {}) {
+      return setWhatProducedExternalWork(iThis, &T::acquire, &T::produce, iLabel);
+    }
+
+    template <typename T>
+    auto setWhatProducedExternalWork(T* iThis, const char* iLabel) {
+      return setWhatProducedExternalWork(iThis, es::Label(iLabel));
+    }
+    template <typename T>
+    auto setWhatProducedExternalWork(T* iThis, const std::string& iLabel) {
+      return setWhatProducedExternalWork(iThis, es::Label(iLabel));
+    }
+
+    template <typename T, typename TDecorator>
+    auto setWhatProducedExternalWork(T* iThis, const TDecorator& iDec, const es::Label& iLabel = {}) {
```

Comment:
I think it would be good to add a comment that the `TDecorator` is applied only for the `produce()` function. (or alternatively drop this overload)

Reply:
Done. A good idea.